todo_app/todo_view_model.py

```diff
@@ -0,0 +1,74 @@
+class ToDoViewModel:
+    def __init__(self, items, recent_date):
+        self._items = items
+        self.determine_done_items() 
+        self.determine_recent_done_items(recent_date)
+        self.determine_older_done_items(recent_date)
+        self.determine_show_all_done()
```

Comment:
These three lines shouldn't be in the constructor (along with the `recent_date` argument). 

We shouldn't be trying to cache any values here. Let them be loaded lazily (like you've done with `to_do_items` & `doing_items` below)